include/treelite/tree_impl.h

```diff
@@ -520,120 +514,141 @@ Tree::GetCategoricalFeatures() const {
   return result;
 }
 
+template <typename ThresholdType, typename LeafOutputType>
 inline int
-Tree::LeftChild(int nid) const {
+Tree<ThresholdType, LeafOutputType>::LeftChild(int nid) const {
```

Comment:
There are lots of small methods like these in `Tree`, `Model` and `ModelImpl`; I wonder if it would be better to have them inside the class definition. (I don't know the original reason behind having them outside of the class definition, so this is just a suggestion).

Reply:
The `tree.h` header was getting very long, so I moved all implementations to a separate file.

Reply:
Fixed in #201